src/main/java/com/google/api/gax/bundling/ThresholdBundler.java

```diff
@@ -182,56 +188,48 @@ public void flush() {
    *
    * @return the number of items added to 'bundle'.
    */
-  public int drainNextBundleTo(Collection<? super E> outputCollection) {
+  public E takeBundle() {
     final Lock lock = this.lock;
     lock.lock();
     try {
-      Bundle outBundle = null;
+      E outBundle = null;
       if (closedBundles.size() > 0) {
         outBundle = closedBundles.remove(0);
```

Comment:
I didn't notice before. This will take time linear to the size of the list. `closeBundles` should be a `Deque`, not a `List`. We can get to this later; it will become moot after the push-pull restructuring anyway.